compiler/optimizations/gpuTransforms.cpp

```diff
@@ -1140,85 +1205,92 @@ static void generateGpuAndNonGpuPaths(const GpuizableLoop &gpuLoop,
   // will make sure that we call the runtime support as if there's a GPU, yet
   // still executing the loop always.
 
-  if(fGpuSpecialization) {
```

Comment:
At the very least, we need to change the comment. But I am still reading the diff, and trying to understand this change.